api/app/models/comment.rb

```diff
@@ -28,9 +29,63 @@ class Comment < ApplicationRecord
                       counter_cache: :children_count
   has_many :children, class_name: "Comment", foreign_key: :parent_id, dependent: :destroy,
                       inverse_of: :parent, counter_cache: :children_count
-  has_many :flags, as: :flaggable, dependent: :destroy, inverse_of: :flaggable
+  has_one :comment_created_event, -> { where event_type: Event::COMMENT_CREATED },
+          class_name: Event,
+          as: :subject,
+          dependent: :destroy,
+          inverse_of: :subject
+  delegate :project, to: :subject
 
   # Validations
   validates :body, :subject, presence: true
 
+  # Callbacks
+  after_create :notify_parent_creator!, :notify_project_editors!
+  after_commit :trigger_event_creation, on: [:create]
+
+  # rubocop:disable Metrics/LineLength, Metrics/AbcSize
+  def subject_url
+    client_url = Rails.configuration.manifold.url
+
+    if subject.is_a? Resource
+      "#{client_url}/projects/#{subject.project.slug}/resource/#{subject.slug}"
+    elsif subject.is_a? Annotation
+      "#{client_url}/read/#{subject.text.slug}/section/#{subject.text_section_id}#annotation-#{subject.id}"
+    end
+  end
+  # rubocop:enable Metrics/LineLength, Metrics/AbcSize
+
+  private
+
+  def trigger_event_creation
+    Event.trigger(Event::COMMENT_CREATED, self)
+  end
+
+  def notify_parent_creator!
```

Comment:
If I understand it correctly, this implementation seems a little naive. Isn't it only notifying the immediate parent? Shouldn't all parents be notified, or perhaps even everyone who is involved in a thread.

Moreover, this model is the wrong place to be making decisions around whether the parent should be notified. It would expect this to kick off a job. In that job (or in a different service class), you can make decisions about who should be notified. It is not the concern of the comment model to figure this out. In other words, this `notify_parent_creator!` method should be a one-liner, not a method that includes logic.

Reply:
Wherever we land on deciding who gets notified, there should be a spec that clearly describes the notification logic.

Reply:
✔